### PR TITLE
enforce metrics order

### LINKED ISSUE #352: enforce report metric order or fix built in report metric ordering
I'm comparing two reports that each have only MAE, MBE, and RMSE. One is a report I created for table mountain ghi that includes nighttime. The other is the built in reference data report. The report I created displays metrics as MAE, MBE, RMSE. The built in report displays MBE, MAE, RMSE. It makes it hard to flip back and forth between the two tabs.

### PR DESCRIPTION
**Pull request by wholmgren**
- [x] Closes #352  .
  - [x] I am familiar with the [contributing guidelines](https://solarforecastarbiter-core.readthedocs.io/en/latest/contributing.html).
  - [x] Tests added.
  - [x] Updates entries to [`docs/source/api.rst`](https://github.com/SolarArbiter/solarforecastarbiter-core/blob/master/docs/source/api.rst) for API changes.
  - [x] Adds descriptions to appropriate "what's new" file in [`docs/source/whatsnew`](https://github.com/SolarArbiter/solarforecastarbiter-core/tree/master/docs/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
  - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
  - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

### CODE REVIEW

**Comment by wholmgren** [comment]
tests pass locally

File: solarforecastarbiter/metrics/calculator.py [hunk 1]
```diff
@@ -249,11 +249,45 @@ def calculate_deterministic_metrics(processed_fx_obs, categories, metrics,
...
+    -------
+    tuple
+        Sorted elements. Sorting order is:
+            * Category
+            * Metric
+            * Index
+            * Value
+    """
+    metric_ordering = list(mapping.keys())
+    category_ordering = list(datamodel.ALLOWED_CATEGORIES.keys())
```
Comment (alorenzo175) [inline solarforecastarbiter/metrics/calculator.py:277]: I feel like we should comment that this ordering is only necessarily preserved in python >= 3.6 with dicts ordered by default now

File: docs/source/whatsnew/1.0.0b5.rst [hunk 2]
```diff
@@ -97,6 +97,7 @@ Bug fixes
 * Fix type errors in observation validation by ensuring all time series values from
   :py:mod:`solarforecastaribter.io.api` functions are coerced to float
   and converting to float before validation (:issue:`252`) (:pull:`384`)
+* Enforce metric order consistency. (:issue:`352`)
```
Comment (alorenzo175) [inline docs/source/whatsnew/1.0.0b5.rst:100]:
```suggestion
* Enforce metric order consistency. (:issue:`352`) (:pull:`396`)
```

File: solarforecastarbiter/metrics/calculator.py (hunk 1 above)
Comment (wholmgren) [reply solarforecastarbiter/metrics/calculator.py:277]: even though we're 3.7+ and use dataclasses?

Comment (alorenzo175) [reply solarforecastarbiter/metrics/calculator.py:277]: ah forgot about the dataclasses that will break 3.6 anyway, so not necessary

**Pull request merged**